share/spack/setup-env.fish

```diff
@@ -788,8 +788,7 @@ end
 #
 set -l fish_version (string split '.' $FISH_VERSION)
 if test $fish_version[1] -gt 3
-    or test $fish_version[1] -eq 3
-    and test $fish_version[2] -ge 2
+    or test $fish_version[1] -eq 3 -a test $fish_version[2] -ge 2
```

Comment:
I think the fish way of doing this is by using `begin` and `end` to group the second two:
> For even more complex conditions, use [begin](https://fishshell.com/docs/current/cmds/begin.html) and [end](https://fishshell.com/docs/current/cmds/end.html) to group parts of them.

```suggestion
    or begin
        test $fish_version[1] -eq 3
        and test $fish_version[2] -ge 2
    end
```
Or on one line
```suggestion
    or begin ; test $fish_version[1] -eq 3 and test $fish_version[2] -ge 2 ; end
```

Ref: https://fishshell.com/docs/current/tutorial.html#conditionals-if-else-switch

Reply:
TIL. That section is very easy to miss.

Fixed by [f59ffe0](https://github.com/spack/spack/pull/48806/commits/f59ffe0ee3a5d6a54f5903183137f887903ff974)